engine/lib/widgets.php

```diff
@@ -303,46 +305,50 @@ function _elgg_create_default_widgets($event, $type, $entity) {
 	// event is already guaranteed by the hook registration.
 	// need to check subtype and type.
 	foreach ($default_widget_info as $info) {
-		if ($info['entity_type'] == $type) {
-			if ($info['entity_subtype'] == ELGG_ENTITIES_ANY_VALUE || $info['entity_subtype'] == $subtype) {
-				// need to be able to access everything
-				$old_ia = elgg_set_ignore_access(true);
-				elgg_push_context('create_default_widgets');
-
-				// pull in by widget context with widget owners as the site
-				// not using elgg_get_widgets() because it sorts by columns and we don't care right now.
-				$options = [
-					'type' => 'object',
-					'subtype' => 'widget',
-					'owner_guid' => elgg_get_site_entity()->guid,
-					'private_setting_name' => 'context',
-					'private_setting_value' => $info['widget_context'],
-					'limit' => 0
-				];
-
-				$widgets = elgg_get_entities_from_private_settings($options);
-				/* @var \ElggWidget[] $widgets */
+		if ($info['entity_type'] !== $type) {
+			continue;
+		}
+		
+		if ($info['entity_subtype'] !== ELGG_ENTITIES_ANY_VALUE && $info['entity_subtype'] !== $subtype) {
+			continue;
+		}
+		
+		// need to be able to access everything
+		$old_ia = elgg_set_ignore_access(true);
+		elgg_push_context('create_default_widgets');
 
-				foreach ($widgets as $widget) {
-					// change the container and owner
-					$new_widget = clone $widget;
-					$new_widget->container_guid = $entity->guid;
-					$new_widget->owner_guid = $entity->guid;
+		// pull in by widget context with widget owners as the site
+		// not using elgg_get_widgets() because it sorts by columns and we don't care right now.
+		$options = [
+			'type' => 'object',
+			'subtype' => 'widget',
+			'owner_guid' => elgg_get_site_entity()->guid,
+			'private_setting_name' => 'context',
+			'private_setting_value' => $info['widget_context'],
+			'limit' => 0
+		];
 
-					// pull in settings
-					$settings = get_all_private_settings($widget->guid);
+		$widgets = elgg_get_entities_from_private_settings($options);
+		/* @var \ElggWidget[] $widgets */
 
-					foreach ($settings as $name => $value) {
-						$new_widget->$name = $value;
-					}
+		foreach ($widgets as $widget) {
+			// change the container and owner
+			$new_widget = clone $widget;
+			$new_widget->container_guid = $entity->guid;
+			$new_widget->owner_guid = $entity->guid;
 
-					$new_widget->save();
-				}
+			// pull in settings
+			$settings = get_all_private_settings($widget->guid);
```

Comment:
why isn't this part of `ElggWidget::clone`

Reply:
yep should... not going to fix that as part of this PR